packages/editorconfig/src/browser/editorconfig-document-manager.ts

```diff
@@ -62,25 +69,18 @@ export class EditorconfigDocumentManager {
         const monacoEditor = MonacoEditor.get(editorWidget);
         if (monacoEditor) {
             monacoEditor.document.onWillSaveModel(event => {
-                event.waitUntil(new Promise<monaco.editor.IIdentifiedSingleEditOperation[]>(resolve => {
+                event.waitUntil(new Promise<monaco.editor.IIdentifiedSingleEditOperation[]>(async resolve => {
+                    const edits: monaco.editor.IIdentifiedSingleEditOperation[] = [];
                     const uri = monacoEditor.uri.toString();
                     const properties = this.properties[uri];
 
-                    const edits = [];
-                    edits.push(...this.getEditsTrimmingTrailingWhitespaces(monacoEditor, properties, event.reason));
+                    // actually format the document (doesn't return an operation)
+                    await this.formatDocument(monacoEditor);
 
+                    // gather operations
+                    edits.push(...await this.getEditsTrimmingTrailingWhitespaces(monacoEditor, properties, event.reason));
                     const edit = this.getEditInsertingFinalNewLine(monacoEditor, properties);
-                    if (edit) {
-                        edits.push(edit);
-
-                        // get current cursor position
-                        const cursor = monacoEditor.cursor;
-
-                        // and then restore it after resolving the promise
-                        setTimeout(() => {
-                            monacoEditor.cursor = cursor;
-                        }, 0);
-                    }
+                    if (edit) { edits.push(edit); }
```

Comment:
this should be broken into multiple lines :)

Reply:
Agree, Done